File: code/lista4solver/exec10.py

# -*- coding: utf-8 -*-
"""
10)
Construa um programa que seja capaz de concluir qual dentre os animais seguintes 
foi escolhido através de perguntas e respostas. Animais possíveis: leão, cavalo, homem, 
macaco, baleia, avestruz, pingüim, pato, águia, tartarug
a, crocodilo e cobra

Exemplo:
É mamífero: Sim
É quadrúpede: Sim
É carnívoro: Não
É herbívoro: Sim
Então o animal escolhido foi o cavalo
"""
quest = str(raw_input(" Mamifero ? "))
if quest=="y":
 quest2 = str(raw_input(" quadrupede ? "))
 if quest2=="y":
  quest3 = str(raw_input(" carnivoro ? "))
  if quest3 == "y":
        print "Leão"
  else:
   quest3 = str(raw_input(" herbivoro ? "))
   if quest3 == "y":
        print "Cavalo"
 else:
    quest2 = str(raw_input("bipede ?"))
    if quest2 == 'y':
        quest3 = str(raw_input("onivoro ?"))
        if quest3=='y':
            print 'Humano'
        else:
            quest3 = str(raw_input("frutifero ?"))
            if quest3=='y':
              print 'macaco'
    else:
        voa = str(raw_input("voador ?"))
        if voa=='y':
            print "Morcego"
        else:
            nada = str(raw_input("nada ?"))
            if nada == 'y':
                print "Baleia"
                        
else:
    ave = str(raw_input(" Ave ? "))
    if ave=="y":
        voa = str(raw_input("voa ?"))
        if voa == "y":
            nada = str(raw_input("nadadora ?"))
            if nada=='y':
                print "Pato"
            else:
                rapina = str(raw_input("rapina ?"))
                if rapina=='y':
                  print "Aguia"
                
        else:
            tropical = str(raw_input("Tropical ?"))
            if tropical == "y":
                print 'Avestruz'
            else:
                polar = str(raw_input("polar ?"))
                if polar == "y": 
                 print "Pinguim"       
    else:
        reptil = str(raw_input("Reptil ? "))
        if reptil=='y':
            casco = str(raw_input("casco ? "))
            if casco =='y':
                print "Tartaruga"
            else:
                carnivoro = str(raw_input("carnivoro ?"))
                if carnivoro =='y':
                    print "Crocodilo"
                else:
                    print "cobra"

File: code/lista4solver/exec10.c
#include <stdio.h>
#include <stdlib.h>


int main(){
	int opcao;
    int move;
	int come;
	int massa;

    printf(" 0 - mamifero  \n 1 - ave\n 2 - reptil \n3 - SAIR\nNumero => ");
    scanf("%d",&opcao);
    if (opcao == 0){
    	printf(" 0 - quadrupede  \n 1 - bipede\n 2 - voa \n 3 - nada\nNumero => ");
        scanf("%d",&move);
        if (move == 0){
        	printf("0 - carnivoro \n1 - herbivoro\nNumero =>" );
        	scanf("%d",&come);
        	switch(come){
        		case 0:
        		printf("leão\n");
        		break;
        		case 1:
        		printf("cavalo\n");
        		break;

        	}
        }

        if (move==1){
        	printf("0 - onivoro \n1 - frutivoro\nNumero => ");
	       	scanf("%d",&come);
	       	switch(come){
	       		case 0:
	       		printf("humano\n");
	       		break;
	       		case 1:
	       		printf("macaco\n");
	       		break;

        	}

        }if (move == 2){
			printf("Morcego\n");
        }if (move == 3){
        	printf("Baleia\n");
        }
    }
    if (opcao ==  1){
    	printf("0 - voa\n1 - nada\n2 - não voa\n \nNumero => ");
    	scanf("%d",&move);
    	switch(move){
    		case 0:
    		  printf("aguia\n");
    		  break;

    		case 1:
    		 printf("pato\n");
    		 break;
    		case 2:
    		 printf("0 - Tropical \n1 - polar\nNumero => ");
    		 scanf("%d",&massa);
    		 if (massa == 0)
    		    printf("Avestruz\n");
    		 if (massa == 1)
    		 	printf("Pinguim\n");
    		 break;
    	}
         
    }
    if (opcao == 2){
    	printf("0 - com casco\n1 - carnivoros\n2 - sem patas\n Numero => ");
        int anda;
        scanf("%d",&anda);
    	switch(anda){
    		case 0:
    		  printf("tartaruga \n");
    		  break;
    		case 1:
    		  printf("crocodilo \n");
    		  break;
    		case 2:
    		  printf("cobra \n");
    		  break;

    	}
    }
    if (opcao == 3){
    	printf("SAIR\n");
    }
    

return 0;
}



 